### PR TITLE
WIP feat:  added support for Centos 9 Stream

### LINKED ISSUE #2033: csi-rbdplugin pods remains in "CrashLoopBackOff" due to old version of modprobe in ceph container image
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

csi-rbdplugin pods should not remains in "CrashLoopBackOff" after the installation of rook-ceph

**Expected behavior:**

csi-rbdplugin pods should come up fine after the installation of rook-ceph

**How to reproduce it (minimal and precise):**

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

cs-rbdplugin pod log is at https://gist.github.com/Rajpratik71/48ef27a62f16834266c23fce441f9476

**Environment**:
* OS (e.g. from /etc/os-release):
`
NAME="SLES"
VERSION="15-SP4"
VERSION_ID="15.4"
PRETTY_NAME="SUSE Linux Enterprise Server 15 SP4"
ID="sles"
ID_LIKE="suse"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:15:sp4"
DOCUMENTATION_URL="https://documentation.suse.com/"
`

* Kernel (e.g. `uname -a`): 
`
Linux xxxxxxxxx 5.14.21-150400.22-default #1 SMP PREEMPT_DYNAMIC Wed May 11 06:57:18 UTC 2022 (49db222) x86_64 x86_64 x86_64 GNU/Linux
`

* Cloud provider or hardware configuration:
Libvirt/KVM based VM

* Rook version (use `rook version` inside of a Rook Pod):
`
rook: v1.9.2
`

* Storage backend version (e.g. for ceph do `ceph -v`):
ceph version 16.2.7 (dd0603118f56ab514f133c8d2e3adfc983942503) pacific (stable)

* Kubernetes version (use `kubectl version`):
`
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.10", GitCommit:"a7a32748b5c60445c4c7ee904caf01b91f2dbb71", GitTreeState:"clean", BuildDate:"2022-02-16T11:18:16Z", GoVersion:"go1.16.14", Compiler:"gc", Platform:"linux/amd64"}
`

* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
Virtualized cluster on LibVirt/KVM Based VM

* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest/Troubleshooting/ceph-toolbox/#interactive-toolbox)):

`
HEALTH_WARN 1 MDSs report slow metadata IOs; clock skew detected on mon.c; Reduced data availability: 96 pgs inactive; 3976 slow ops, oldest one blocked for 4558 sec, mon.c has slow ops; OSD count 2 < osd_pool_default_size 3
`

### PR DESCRIPTION
**Pull request by Rajpratik71**
Description of your changes:
WIP feat:  added support for Centos 9 Stream

Which issue is resolved by this Pull Request:
Resolves #2033 

Checklist:
- [*] Documentation has been updated, if necessary.
- [*] Pending release notes updated with breaking and/or notable changes, if necessary.

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

### CODE REVIEW

**Comment by Rajpratik71** [comment]
This is one other approach I was trying before #2037 .

Here, I got stuck due to a lot of templating done for extracting the CentOS version and using that in URLs.

And I think, the issue of the missing packages will be faced here which happened in #2037

**Comment by prazumovsky** [comment]
Does this PR blocked? There are no updates for a long time.

**Comment by calvinxu** [comment]
How about this PR progress, when it is expected to be merged for resolving this issue https://github.com/ceph/ceph-container/issues/2033? Suppose by then Ceph-CSI project to use the  CentOS Stream 9 as it's OS container-image?

**Comment by nixpanic** [comment]
@Rajpratik71 or @djgalloway, could you let us know what still needs to be done for this PR? There is a lot of interest in this, maybe someone else can assist with finalizing the work.

File: ceph-releases/ALL/centos/9-stream/daemon-base/__DOCKERFILE_POSTINSTALL_CLEANUP__ [hunk 1]
```diff
@@ -0,0 +1,4 @@
+echo 'Postinstall cleanup' && \
+ (rm -rf "/usr/bin/hyperkube /usr/bin/etcd /usr/bin/systemd-analyze /usr/share/hwdata/{iab.txt,oui.txt} /etc/profile.d/lang.sh" && \
+   yum clean all && \
```
Comment (nixpanic) [inline ceph-releases/ALL/centos/9-stream/daemon-base/__DOCKERFILE_POSTINSTALL_CLEANUP__:3]: consider using the `dnf` command, `yum` still works, but redirects to `dnf`

**Review by nixpanic — COMMENTED** [review]
The CI job fails in the **switch to podman** step. Podman is already installed on recent GitHub runners with ubuntu-latest. There is no need to install podman a 2nd time.

**Comment by djgalloway** [comment]
> @Rajpratik71 or @djgalloway, could you let us know what still needs to be done for this PR? There is a lot of interest in this, maybe someone else can assist with finalizing the work.

I'm sorry but I'm no longer working on the Ceph project so I'm not sure what the path to get this merged might look like.

**Comment by Rajpratik71** [comment]
> @Rajpratik71 or @djgalloway, could you let us know what still needs to be done for this PR? There is a lot of interest in this, maybe someone else can assist with finalizing the work.

@nixpanic  you can assume current changes as base changes which need more modification and testing for CentOS 9 support.

**Comment by nixpanic** [comment]
#2183 seems to have been a replacement for this. @guits can probably confirm and close this PR.

**Pull request closed without merging**